buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1258,23 +1258,20 @@ class BeamModulePlugin implements Plugin<Project> {
         exclude group: "org.hamcrest", module: "hamcrest-all"
       }
 
-      // Force usage of the libraries defined within our common set found in the root
+      // For tests, force usage of the libraries defined within our common set found in the root
       // build.gradle instead of using Gradles default dependency resolution mechanism
       // which chooses the latest version available.
       //
       // TODO: Figure out whether we should force all dependency conflict resolution
-      // to occur in the "shadow" and "shadowTest" configurations.
+      // to occur in the "shadowTest" configurations.
       project.configurations.all { config ->
-        // When running beam_Dependency_Check, resolutionStrategy should not be used; otherwise
-        // gradle-versions-plugin does not report the latest versions of the dependencies.
-        def startTasks = project.gradle.startParameter.taskNames
-        def inDependencyUpdates = 'dependencyUpdates' in startTasks || 'runBeamDependencyCheck' in startTasks
-
-        // The "errorprone" configuration controls the classpath used by errorprone static analysis, which
-        // has different dependencies than our project.
-        if (config.getName() != "errorprone" && !inDependencyUpdates) {
+        String configName = config.getName()
+        if (configName.toLowerCase().contains('test')) {
           config.resolutionStrategy {
-            force project.library.java.values()
+            force project.library.java.junit,
+                    project.library.java.hamcrest_core,
+                    project.library.java.guava, // sdks:java:io:google-cloud-platform needs this
```

Comment:
If its only needed in these projects, does it make sense to apply force rule there?

Reply:
I find centrally managing `force` more clear.